e2e/tests/wasm/grandpa_test.go

```diff
@@ -135,79 +145,110 @@ func (s *GrandpaTestSuite) TestMsgTransfer_Succeeds_GrandpaContract() {
 	// Start relayer
 	s.Require().NoError(r.StartRelayer(ctx, eRep, pathName))
 
-	// TODO: this can be refactored to broadcast a MsgTransfer instead of CLI.
-	// https://github.com/cosmos/ibc-go/issues/4963
-	// Send 1.77 stake from cosmosUser to parachainUser
-	amountToSend := int64(1_770_000)
-	transfer := ibc.WalletAmount{
-		Address: polkadotUser.FormattedAddress(),
-		Denom:   cosmosChain.Config().Denom,
-		Amount:  math.NewInt(amountToSend),
-	}
-	tx, err := cosmosChain.SendIBCTransfer(ctx, "channel-0", cosmosUser.KeyName(), transfer, ibc.TransferOptions{})
-	s.Require().NoError(err)
-	s.Require().NoError(tx.Validate()) // test source wallet has decreased funds
-	err = testutil.WaitForBlocks(ctx, 15, cosmosChain, polkadotChain)
-	s.Require().NoError(err)
-
-	// Verify tokens arrived on parachain user
-	parachainUserStake, err := polkadotChain.GetIbcBalance(ctx, string(polkadotUser.Address()), 2)
-	s.Require().NoError(err)
-	s.Require().Equal(amountToSend, parachainUserStake.Amount.Int64(), "parachain user's stake amount not expected after first tx")
-
-	// Send 1.16 stake from parachainUser to cosmosUser
-	amountToReflect := int64(1_160_000)
-	reflectTransfer := ibc.WalletAmount{
-		Address: cosmosUser.FormattedAddress(),
-		Denom:   "2", // stake
-		Amount:  math.NewInt(amountToReflect),
-	}
-	_, err = polkadotChain.SendIBCTransfer(ctx, "channel-0", polkadotUser.KeyName(), reflectTransfer, ibc.TransferOptions{})
-	s.Require().NoError(err)
-
-	// Send 1.88 "UNIT" from Alice to cosmosUser
-	amountUnits := math.NewInt(1_880_000_000_000)
-	unitTransfer := ibc.WalletAmount{
-		Address: cosmosUser.FormattedAddress(),
-		Denom:   "1", // UNIT
-		Amount:  amountUnits,
-	}
-	_, err = polkadotChain.SendIBCTransfer(ctx, "channel-0", "alice", unitTransfer, ibc.TransferOptions{})
-	s.Require().NoError(err)
-
-	// Wait for MsgRecvPacket on cosmos chain
-	finalStakeBal := math.NewInt(fundAmount - amountToSend + amountToReflect)
-	err = cosmos.PollForBalance(ctx, cosmosChain, 20, ibc.WalletAmount{
-		Address: cosmosUser.FormattedAddress(),
-		Denom:   cosmosChain.Config().Denom,
-		Amount:  finalStakeBal,
+	t.Run("IBC transfer from Cosmos chain to Polkadot parachain times out", func(t *testing.T) {
```

Comment:
did we want to add this as a completely separate test? 

Right now this is being added to  TestMsgTransfer_Succeeds_GrandpaContract, right? This is maybe a little misleading. What do you think?

Reply:
I did it as part of this test bc the setup takes so long for these tests (due to gov proposal needed to create a wasm client, etc).

If you think it's worth it to add as a completely separate test I can do it, but tbh I really don't know if it's worth the setup time, as the functionality can easily be tested in the transfer flow. I could update the test name for clarity, as another option.

Reply:
I agree with you that the test setup is pretty long winded and takes up quite a bit of time. But I think it would just be at the expense of one extra runner, right? 

General separation of concerns would have me leaning towards a separate test, even tho there's also probably some code duplication. I'd be fine with leaving it in this one too if there's consensus, but maybe we could rename the test if that were the case.

Any thoughts? cc. @chatton @DimitrisJim @srdtrk

Reply:
I'd lean towards a separate test too, in most cases these tests will be running side by side anyway, so the time difference shouldn't be noticeable.

Reply:
alright! i can make a separate test, no worries. I'd rather do it now than revisit later haha, close it up and zip it up.